src/core/symbology/qgslinesymbollayer.cpp

```diff
@@ -3788,3 +3789,309 @@ void QgsLineburstSymbolLayer::setColorRamp( QgsColorRamp *ramp )
 {
   mGradientRamp.reset( ramp );
 }
+
+//
+// QgsFilledLineSymbolLayer
+//
+
+QgsFilledLineSymbolLayer::QgsFilledLineSymbolLayer( double width, QgsFillSymbol *fillSymbol )
+  : QgsLineSymbolLayer()
+{
+  mWidth = width;
+  mFill.reset( fillSymbol ? fillSymbol : static_cast<QgsFillSymbol *>( QgsFillSymbol::createSimple( QVariantMap() ) ) );
+}
+
+QgsFilledLineSymbolLayer::~QgsFilledLineSymbolLayer() = default;
+
+QgsSymbolLayer *QgsFilledLineSymbolLayer::create( const QVariantMap &props )
+{
+  double width = DEFAULT_SIMPLELINE_WIDTH;
+
+  if ( props.contains( QStringLiteral( "line_width" ) ) )
+  {
+    width = props[QStringLiteral( "line_width" )].toDouble();
+  }
+  else if ( props.contains( QStringLiteral( "outline_width" ) ) )
+  {
+    width = props[QStringLiteral( "outline_width" )].toDouble();
+  }
+  else if ( props.contains( QStringLiteral( "width" ) ) )
+  {
+    //pre 2.5 projects used "width"
+    width = props[QStringLiteral( "width" )].toDouble();
+  }
```

Comment:
Do we need this for a new symbol layer that? We'll never see a pre 2.5 project with this symbol layer type ever :wink: . If you were thinking of adding it as an alias, maybe change the comment?